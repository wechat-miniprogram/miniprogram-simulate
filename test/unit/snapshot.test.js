const simulate = require('../../dist/miniprogram_simulate.cjs.js')

test('snapshot', () => {
  const id = simulate.loadComponentByDef(
    {},
    `
      <abc class="index" bind:tap="onTap">{{prop}}</abc>
      <div class="index">
          <div class="inner">321</div>
      </div>
    `,
    {
      properties: {
        prop: {
          type: String,
          value: 'index.properties',
        },
      },
      data: {
        a: 123,
      },
      methods: {
        onTap() {
          this.setData({
            prop: 'tap',
          })
        },
      },
    }
  )
  const comp = simulate.render(id)
  comp.attach(document.body)
  expect(comp).toMatchSnapshot()
  comp.querySelector('.index').dispatchTapEvent()
  expect(comp).toMatchSnapshot()
  comp.detach()
})
