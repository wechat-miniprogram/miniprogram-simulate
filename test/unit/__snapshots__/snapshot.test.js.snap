// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
"<(virtual):shadow>
  <abc class="index">
    index.properties
  <div class="index">
    <div class="inner">
      321"
`;

exports[`snapshot 2`] = `
"<(virtual):shadow>
  <abc class="index">
    tap
  <div class="index">
    <div class="inner">
      321"
`;
